bson/binary.py

```diff
@@ -191,21 +194,64 @@ class UuidRepresentation:
 """
 
 
+VECTOR_SUBTYPE = 9
+"""BSON binary subtype for densely packed vector data.
+
+.. versionadded:: 4.9
+"""
+
+
 USER_DEFINED_SUBTYPE = 128
 """BSON binary subtype for any user defined structure.
 """
 
 
+class BinaryVectorDtype(Enum):
+    """Datatypes of vector subtype.
+
+    The PACKED_BIT value represents a special case where vector values themselves
+    can only hold two values (0 or 1) but these are packed together into groups of 8,
+    a byte. In Python, these are displayed as ints in range(0,128).
+    """
+
+    INT8 = b"\x03"
+    FLOAT32 = b"\x27"
+    PACKED_BIT = b"\x10"
+
+
+# Map from bytes to enum value, for decoding.
+DTYPE_FROM_HEX = {key.value: key for key in BinaryVectorDtype}
+
+
+@dataclass
+class BinaryVector:
+    """Vector of numbers along with metadata for binary interoperability.
+
+    dtype specifies the data type stored in binary.
```

Comment:
I suggest moving these descriptions to docstrings under each property.

Reply:
Like the following?

```
@dataclass
class BinaryVector:
    """Vector of numbers along with metadata for binary interoperability.

    :param: data: Sequence of numbers representing the mathematical vector.
    :param dtype:  specifies the data type stored in binary.
    :param: padding: specifies the number of bits in the final byte that are to be ignored
      when a vector element's size is less than a byte and the length of the vector is not a multiple of 8.
      """

    data: (Sequence[float | int]
    dtype: BinaryVectorDtype
    padding: Optional[int] = 0
```

Reply:
i.e. Try """ under each var